### ISSUE TITLE
Greenshot not compatible with new Outlook version (beta)

### ISSUE THREAD

**Issue by Johnsinoo** [greenshot/greenshot#534]
We at our organization received a complaint that the ‘Microsoft Outlook ’ shortcut in the greenshot menu is not working with the newest version of Outlook. Perhaps this is because the outlook process changed names from Outlook.exe to olk.exe in the newest version?

I’d kindly want to request it as a feature in an upcoming version of Greenshot.

Example steps to reproduce the behavior:
1. Open Outlook and press the button top right to switch to the newest version of Outlook.
2. Make a screenshot with Greenshot and select the 'Microsoft Outlook' function in the menu.
3. You'll notice it's trying it's best to open the screenshot in a new mail, but visually, it's not doing anything. For X-amount of seconds, you are not able to use Greenshot.

When you normally select the 'Microsoft Outlook' function in the menu, it should open the image inside a new e-mail.

The stable version of Greenshot 1.2.10.6 has this problem. I then I tried with the latest version of Greenshot 1.3.277 and it did also not work. This is both an issue on Windows 10 and Windows 11 PC's.

**Comment by Omarlilyo** [comment]
Was this ever solved?

**Comment by drizzt09** [comment]
This is still an issue and New Outlook is more usable now and more are starting to roll to it.
Can this get some support please?

**Comment by Omarlilyo** [comment]
Yep - still an issue here too, and all my staff have moved to new outlook - so i can no longer get them to use greenshot and they are now using windows snipper - which is worse.

**Comment by reconsierra** [comment]
Just adding my vote to this as we have just rolled to the new Outlook globally and this was a disappointing surprise as I used this a lot.

Thanks :)

**Comment by danrhodes** [comment]
I'll take a look later should just need the new version adding in of it's not there

**Comment by Omarlilyo** [comment]
Thanks @danrhodes

**Comment by Lakritzator** [comment]
@danrhodes I am not sure if we need to do something extra...
For me it works for me with the current 1.4 builds, but I am not sure what Outlook version we are having issues with...

(I need to check all the Office apps, if these still work...)

**Comment by danrhodes** [comment]
I'll have a look to see if it's working already on a later build.

**Comment by Omarlilyo** [comment]
@danrhodes  @Lakritzator  

There is "classic" outlook, and "new" outlook.

In typical microsoft fashion, there is very little difference in the icons and they are very easy to confuse.

My current greenshot 1.3.312 (and every one before that) only lets us post out to "classic" outlook. Not the New one.

https://support.microsoft.com/en-gb/office/getting-started-with-the-new-outlook-for-windows-656bb8d9-5a60-49b2-a98b-ba7822bc7627

https://www.reddit.com/r/Office365/comments/1ni74hj/new_outlook_is_a_disaster_why_do_they_persist_it/

issue is - most poeple dont have the option to switch back to old outlook - its decided by their organisation.

**Comment by danrhodes** [comment]
Agreed, we use new outlook at work so will check and I will create a pr to fix if there is an issue.

**Comment by reconsierra** [comment]
> I'll have a look to see if it's working already on a later build.

Thanks a lot Dan - appreciate the fast response.

FYI, my version:
Microsoft Outlook Version: 1.2026.114.100
Client Version: 20260116004.03
WebView2 Version: 144.0.3719.93 (Stable)
Session ID: 0409c907-16a9-be49-b65b-e2587e445054

**Comment by danrhodes** [comment]
An update on this.

There is a bug in the logic that detects if Outlook Classic is installed, or not, and even if it isn't displays the Outlook button.

Hence when clicked it displays the MAPI Error box as the button should not be there in the first place.

I will post a PR to fix this particular issue. Yes I know it's removing the button but it is a bug and shouldn't show the button if Outlook isn't installed.

In terms of sharing to New Outlook this is way way more complex as it is essentially a web app and doesn't support COM we can't just create a new email and attach the image.

I think the solution may lay in the Windows Share method which will allow GS to leverage the Share dialog to essentially share the screenshot with anything on the pc, including connected phones, other apps etc.

I have tested a version of this and it works, but there are many kinks for me to iron out before this will be ready.

I'll carry on tomorrow evening and post a few screenshots.

**Comment by Lakritzator** [comment]
@danrhodes the share is already available, I even fixed it recently and noticed that Microsoft even added bitmap editing capabilities in it 😆

Should be available by default, since I resolved the Win10 plugin into the main Greenshot application... It's in the destination picker, under share.

**Comment by drizzt09** [comment]
> @danrhodes the share is already available, I even fixed it recently and noticed that Microsoft even added bitmap editing capabilities in it 😆
> 
> Should be available by default, since I resolved the Win10 plugin into the main Greenshot application... It's in the destination picker, under share.

Doesn't seem to work with windows 11. See a flicker then gone. 

Since win10 is no longer a supported product. Probably best to move on from it anyways.

**Comment by danrhodes** [comment]
It's not something I use in all honesty, but, if the share option is already there as a button this _should_ solve the issue I would think once we fix the logic that will stop the Outlook button showing if it isn't installed. We can direct users to use the share mechanism instead which from testing earlier worked well to compose an email in new Outlook and attach the screenshot.

TL/DR - this could be a case of just using the share button and fixing the big that incorrectly shows the Outlook button.

I will test.

**Comment by danrhodes** [comment]
> > @danrhodes the share is already available, I even fixed it recently and noticed that Microsoft even added bitmap editing capabilities in it 😆
> > 
> > Should be available by default, since I resolved the Win10 plugin into the main Greenshot application... It's in the destination picker, under share.
> 
> Doesn't seem to work with windows 11. See a flicker then gone. 
> 
> Since win10 is no longer a supported product. Probably best to move on from it anyways. 

I changed the code earlier to reference the Windows 11 share dialog, from which you can select New Outlook and this worked perfectly I'll do more testing tomorrow with the method @Lakritzator mentioned.

@Lakritzator is there a share button in the editor window, it's not something I've ever looked for as in my day to day all I ever need is for the image to be copied to my clipboard.

**Comment by Lakritzator** [comment]
Not sure, in theory it could be available as the share destination in the editor too yeah... I believe all/most destinations are available in both the destination picker and the editor.

@drizzt09 I only recently fixed it, so you would need to be on a pre-release of 1.4

**Comment by danrhodes** [comment]
I can see "Windows 10 Share", but it doesn't seem to do anything on Windows 11?

Given Windows 10 is now EOL, should we rename this to Windows Share (remove the OS ver for future proofing) and ensure it works on Windows 11 and Windows 10 for backwards compatibility?

<img width="864" height="302" alt="Image" src="https://github.com/user-attachments/assets/64286afc-b5a2-4b79-bd36-77f1a72b5fda" />

<img width="578" height="270" alt="Image" src="https://github.com/user-attachments/assets/a5f9cc9d-ccd9-419b-a56b-32b78e8fa2fe" />

**Comment by danrhodes** [comment]
Ok to give context to the two PRs - I have created one PR https://github.com/greenshot/greenshot/pull/919 to get the Share Button working in Windows 11, also changed the name for Windows Share and Windows OCR while I was in the same area of code to align with Windows 10 now being EOL. Don't panic, it still works with Windows 10 for those users still on old OS's

I have also created PR https://github.com/greenshot/greenshot/pull/920 to not show the Outlook button if the device doesnt have Outlook Classic installed which can A: be confusing for the user and B: cause the user to receive a MAPI Error.

**Comment by danrhodes** [comment]
> Not sure, in theory it could be available as the share destination in the editor too yeah... I believe all/most destinations are available in both the destination picker and the editor.
> 
> @drizzt09 I only recently fixed it, so you would need to be on a pre-release of 1.4

I tested today on two individual Win11 machines on 1.4.88 and confirmed the Windows 10 Share feature did not work on Windows 11 hence the PR which fixes it to work on both 👍

**Comment by Omarlilyo** [comment]
Was this ever solved? I saw a new version was out. TY!

**Comment by danrhodes** [comment]
> Was this ever solved? I saw a new version was out. TY!

Give it a try it doesn't share directly to outlook but instead to windows share which will then be able to share with outlook.

**Comment by Omarlilyo** [comment]
just tested- still portsout to "outlook" "classic" Which suits me,but not my team :D the option for one or the other would be amazing.

**Comment by danrhodes** [comment]
I test on the pr and was able to select email which sent to my default mail client new outlook.

Unfortunately due to new outlook being a web app and using Microsoft graph it's not directly possible to share to 👎

**Comment by traywolf** [comment]
I too am getting a MAPI error, apparently due to Outlook.  The error is barely a concern.
Rhetorical comment: 
My workstation uses Office O365, with Outlook Classic. I do not, nor have I ever configured Outlook as a mail client.  Perhaps an option to tell Greenshot to not even look?  

Nice product.  Donated twice in the past.  Thrilled you wrote this.  use it daily.
